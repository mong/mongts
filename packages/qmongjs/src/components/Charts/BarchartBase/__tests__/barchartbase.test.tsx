import React from "react";
import { render } from "@testing-library/react";
import { BarchartBase } from "..";
import { vi, test, expect } from "vitest";
import {
  IndicatorData,
  DataPoint,
} from "../../../IndicatorTable/IndicatortablebodyV2";

// eslint-disable-next-line @typescript-eslint/no-require-imports
vi.mock("next/router", () => require("next-router-mock"));
vi.mock("next/font/google", () => ({
  Plus_Jakarta_Sans: () => ({
    weight: ["200", "300", "400", "500", "600", "700", "800"],
    subsets: ["latin"],
    display: "swap",
    style: {
      fontFamily: "Plus Jakarta Sans",
    },
  }),
}));

const barchartTestData = {
  indicatorID: "test",
  indicatorName: "Test",
  levelGreen: 0.8,
  levelYellow: 0.6,
  levelDirection: 1,
  minDenominator: 5,
  shortDescription: "Test",
  longDescription: "Test",
  sortingName: "test",
  data: [
    {
      unitName: "Tromsø",
      year: 2023,
      var: 0.65,
      denominator: 30,
      format: null,
      level_direction: 1,
      level_green: 0.8,
      level_yellow: 0.6,
      dg: null,
    },
    {
      unitName: "Bodø",
      year: 2023,
      var: 0.7,
      denominator: 30,
      format: null,
      level_direction: 1,
      level_green: 0.8,
      level_yellow: 0.6,
      dg: null,
    },
    {
      unitName: "Harstad",
      year: 2023,
      var: 0.8,
      denominator: 30,
      format: null,
      level_direction: 1,
      level_green: 0.8,
      level_yellow: 0.6,
      dg: null,
    },
  ] as DataPoint[],
} as IndicatorData;

test("Standard render", () => {
  const { container } = render(
    <BarchartBase
      indicatorData={barchartTestData}
      width={800}
      height={800}
      xTickFormat={",.0%"}
    />,
  );
  expect(container).toMatchSnapshot();
});
