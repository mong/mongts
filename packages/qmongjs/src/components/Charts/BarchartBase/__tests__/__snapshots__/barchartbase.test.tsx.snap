// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Standard render 1`] = `
<div>
  <svg
    height="880"
    width="990"
  >
    <rect
      fill="#FFFFFF"
      height="880"
      rx="14"
      width="990"
      x="0"
      y="0"
    />
    <g
      class="visx-group"
      transform="translate(0, 0)"
    >
      <g
        class="visx-group"
        transform="translate(150, 40)"
      >
        <rect
          class="visx-bar"
          fill="#BFCED6"
          height="200.00000000000003"
          rx="4"
          width="520"
          x="0"
          y="50"
        />
        <rect
          class="visx-bar"
          fill="#BFCED6"
          height="200.00000000000003"
          rx="4"
          width="560"
          x="0"
          y="300"
        />
        <rect
          class="visx-bar"
          fill="#BFCED6"
          height="200.00000000000003"
          rx="4"
          width="640"
          x="0"
          y="550"
        />
      </g>
      <g
        class="visx-group visx-axis visx-axis-left"
        transform="translate(150, 40)"
      >
        <g
          class="visx-group visx-axis-tick"
          transform="translate(0, 0)"
        >
          <line
            class="visx-line"
            fill="transparent"
            shape-rendering="crispEdges"
            stroke="#000000"
            stroke-linecap="square"
            stroke-width="1"
            x1="0"
            x2="-8"
            y1="150"
            y2="150"
          />
          <svg
            font-size="11"
            style="overflow: visible;"
            x="-0.25em"
            y="0.33em"
          >
            <text
              fill="#000000"
              font-family="Arial"
              font-size="11"
              text-anchor="end"
              transform=""
              x="-8"
              y="150"
            >
              <tspan
                dy="0em"
                x="-8"
              >
                Tromsø
              </tspan>
            </text>
          </svg>
        </g>
        <g
          class="visx-group visx-axis-tick"
          transform="translate(0, 0)"
        >
          <line
            class="visx-line"
            fill="transparent"
            shape-rendering="crispEdges"
            stroke="#000000"
            stroke-linecap="square"
            stroke-width="1"
            x1="0"
            x2="-8"
            y1="400"
            y2="400"
          />
          <svg
            font-size="11"
            style="overflow: visible;"
            x="-0.25em"
            y="0.33em"
          >
            <text
              fill="#000000"
              font-family="Arial"
              font-size="11"
              text-anchor="end"
              transform=""
              x="-8"
              y="400"
            >
              <tspan
                dy="0em"
                x="-8"
              >
                Bodø
              </tspan>
            </text>
          </svg>
        </g>
        <g
          class="visx-group visx-axis-tick"
          transform="translate(0, 0)"
        >
          <line
            class="visx-line"
            fill="transparent"
            shape-rendering="crispEdges"
            stroke="#000000"
            stroke-linecap="square"
            stroke-width="1"
            x1="0"
            x2="-8"
            y1="650"
            y2="650"
          />
          <svg
            font-size="11"
            style="overflow: visible;"
            x="-0.25em"
            y="0.33em"
          >
            <text
              fill="#000000"
              font-family="Arial"
              font-size="11"
              text-anchor="end"
              transform=""
              x="-8"
              y="650"
            >
              <tspan
                dy="0em"
                x="-8"
              >
                Harstad
              </tspan>
            </text>
          </svg>
        </g>
        <line
          class="visx-line visx-axis-line"
          fill="transparent"
          shape-rendering="crispEdges"
          stroke="#000000"
          stroke-width="1"
          x1="0"
          x2="0"
          y1="0.5"
          y2="800.5"
        />
      </g>
      <g
        class="visx-group visx-axis visx-axis-bottom"
        transform="translate(150, 840)"
      >
        <g
          class="visx-group visx-axis-tick"
          transform="translate(0, 0)"
        >
          <line
            class="visx-line"
            fill="transparent"
            shape-rendering="crispEdges"
            stroke="#000000"
            stroke-linecap="square"
            stroke-width="1"
            x1="0"
            x2="0"
            y1="0"
            y2="8"
          />
          <svg
            font-size="11"
            style="overflow: visible;"
            x="0"
            y="0.25em"
          >
            <text
              fill="#000000"
              font-family="Arial"
              font-size="11"
              text-anchor="middle"
              transform=""
              x="0"
              y="19"
            >
              <tspan
                dy="0em"
                x="0"
              >
                0 %
              </tspan>
            </text>
          </svg>
        </g>
        <g
          class="visx-group visx-axis-tick"
          transform="translate(0, 0)"
        >
          <line
            class="visx-line"
            fill="transparent"
            shape-rendering="crispEdges"
            stroke="#000000"
            stroke-linecap="square"
            stroke-width="1"
            x1="80"
            x2="80"
            y1="0"
            y2="8"
          />
          <svg
            font-size="11"
            style="overflow: visible;"
            x="0"
            y="0.25em"
          >
            <text
              fill="#000000"
              font-family="Arial"
              font-size="11"
              text-anchor="middle"
              transform=""
              x="80"
              y="19"
            >
              <tspan
                dy="0em"
                x="80"
              >
                10 %
              </tspan>
            </text>
          </svg>
        </g>
        <g
          class="visx-group visx-axis-tick"
          transform="translate(0, 0)"
        >
          <line
            class="visx-line"
            fill="transparent"
            shape-rendering="crispEdges"
            stroke="#000000"
            stroke-linecap="square"
            stroke-width="1"
            x1="160"
            x2="160"
            y1="0"
            y2="8"
          />
          <svg
            font-size="11"
            style="overflow: visible;"
            x="0"
            y="0.25em"
          >
            <text
              fill="#000000"
              font-family="Arial"
              font-size="11"
              text-anchor="middle"
              transform=""
              x="160"
              y="19"
            >
              <tspan
                dy="0em"
                x="160"
              >
                20 %
              </tspan>
            </text>
          </svg>
        </g>
        <g
          class="visx-group visx-axis-tick"
          transform="translate(0, 0)"
        >
          <line
            class="visx-line"
            fill="transparent"
            shape-rendering="crispEdges"
            stroke="#000000"
            stroke-linecap="square"
            stroke-width="1"
            x1="240"
            x2="240"
            y1="0"
            y2="8"
          />
          <svg
            font-size="11"
            style="overflow: visible;"
            x="0"
            y="0.25em"
          >
            <text
              fill="#000000"
              font-family="Arial"
              font-size="11"
              text-anchor="middle"
              transform=""
              x="240"
              y="19"
            >
              <tspan
                dy="0em"
                x="240"
              >
                30 %
              </tspan>
            </text>
          </svg>
        </g>
        <g
          class="visx-group visx-axis-tick"
          transform="translate(0, 0)"
        >
          <line
            class="visx-line"
            fill="transparent"
            shape-rendering="crispEdges"
            stroke="#000000"
            stroke-linecap="square"
            stroke-width="1"
            x1="320"
            x2="320"
            y1="0"
            y2="8"
          />
          <svg
            font-size="11"
            style="overflow: visible;"
            x="0"
            y="0.25em"
          >
            <text
              fill="#000000"
              font-family="Arial"
              font-size="11"
              text-anchor="middle"
              transform=""
              x="320"
              y="19"
            >
              <tspan
                dy="0em"
                x="320"
              >
                40 %
              </tspan>
            </text>
          </svg>
        </g>
        <g
          class="visx-group visx-axis-tick"
          transform="translate(0, 0)"
        >
          <line
            class="visx-line"
            fill="transparent"
            shape-rendering="crispEdges"
            stroke="#000000"
            stroke-linecap="square"
            stroke-width="1"
            x1="400"
            x2="400"
            y1="0"
            y2="8"
          />
          <svg
            font-size="11"
            style="overflow: visible;"
            x="0"
            y="0.25em"
          >
            <text
              fill="#000000"
              font-family="Arial"
              font-size="11"
              text-anchor="middle"
              transform=""
              x="400"
              y="19"
            >
              <tspan
                dy="0em"
                x="400"
              >
                50 %
              </tspan>
            </text>
          </svg>
        </g>
        <g
          class="visx-group visx-axis-tick"
          transform="translate(0, 0)"
        >
          <line
            class="visx-line"
            fill="transparent"
            shape-rendering="crispEdges"
            stroke="#000000"
            stroke-linecap="square"
            stroke-width="1"
            x1="480"
            x2="480"
            y1="0"
            y2="8"
          />
          <svg
            font-size="11"
            style="overflow: visible;"
            x="0"
            y="0.25em"
          >
            <text
              fill="#000000"
              font-family="Arial"
              font-size="11"
              text-anchor="middle"
              transform=""
              x="480"
              y="19"
            >
              <tspan
                dy="0em"
                x="480"
              >
                60 %
              </tspan>
            </text>
          </svg>
        </g>
        <g
          class="visx-group visx-axis-tick"
          transform="translate(0, 0)"
        >
          <line
            class="visx-line"
            fill="transparent"
            shape-rendering="crispEdges"
            stroke="#000000"
            stroke-linecap="square"
            stroke-width="1"
            x1="560"
            x2="560"
            y1="0"
            y2="8"
          />
          <svg
            font-size="11"
            style="overflow: visible;"
            x="0"
            y="0.25em"
          >
            <text
              fill="#000000"
              font-family="Arial"
              font-size="11"
              text-anchor="middle"
              transform=""
              x="560"
              y="19"
            >
              <tspan
                dy="0em"
                x="560"
              >
                70 %
              </tspan>
            </text>
          </svg>
        </g>
        <g
          class="visx-group visx-axis-tick"
          transform="translate(0, 0)"
        >
          <line
            class="visx-line"
            fill="transparent"
            shape-rendering="crispEdges"
            stroke="#000000"
            stroke-linecap="square"
            stroke-width="1"
            x1="640"
            x2="640"
            y1="0"
            y2="8"
          />
          <svg
            font-size="11"
            style="overflow: visible;"
            x="0"
            y="0.25em"
          >
            <text
              fill="#000000"
              font-family="Arial"
              font-size="11"
              text-anchor="middle"
              transform=""
              x="640"
              y="19"
            >
              <tspan
                dy="0em"
                x="640"
              >
                80 %
              </tspan>
            </text>
          </svg>
        </g>
        <g
          class="visx-group visx-axis-tick"
          transform="translate(0, 0)"
        >
          <line
            class="visx-line"
            fill="transparent"
            shape-rendering="crispEdges"
            stroke="#000000"
            stroke-linecap="square"
            stroke-width="1"
            x1="720"
            x2="720"
            y1="0"
            y2="8"
          />
          <svg
            font-size="11"
            style="overflow: visible;"
            x="0"
            y="0.25em"
          >
            <text
              fill="#000000"
              font-family="Arial"
              font-size="11"
              text-anchor="middle"
              transform=""
              x="720"
              y="19"
            >
              <tspan
                dy="0em"
                x="720"
              >
                90 %
              </tspan>
            </text>
          </svg>
        </g>
        <g
          class="visx-group visx-axis-tick"
          transform="translate(0, 0)"
        >
          <line
            class="visx-line"
            fill="transparent"
            shape-rendering="crispEdges"
            stroke="#000000"
            stroke-linecap="square"
            stroke-width="1"
            x1="800"
            x2="800"
            y1="0"
            y2="8"
          />
          <svg
            font-size="11"
            style="overflow: visible;"
            x="0"
            y="0.25em"
          >
            <text
              fill="#000000"
              font-family="Arial"
              font-size="11"
              text-anchor="middle"
              transform=""
              x="800"
              y="19"
            >
              <tspan
                dy="0em"
                x="800"
              >
                100 %
              </tspan>
            </text>
          </svg>
        </g>
        <line
          class="visx-line visx-axis-line"
          fill="transparent"
          shape-rendering="crispEdges"
          stroke="#000000"
          stroke-width="1"
          x1="0.5"
          x2="800.5"
          y1="0"
          y2="0"
        />
      </g>
    </g>
  </svg>
</div>
`;
